Moq.AutoMock/Resolvers/EnumerableResolver.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+
+namespace Moq.AutoMock.Resolvers
+{
+    /// <summary>
+    /// A resolver that resolves IEnumerable&lt;T&gt; requested types.
+    /// </summary>
+    public class EnumerableResolver : IMockResolver
+    {
+        /// <summary>
+        /// Resolves IEnumerable&lt;T&gt; types.
+        /// </summary>
+        /// <param name="context">The resolution context.</param>
+        public void Resolve(MockResolutionContext context)
+        {
+            var (am, serviceType, _) = context ?? throw new ArgumentNullException(nameof(context));
+
+            if (!serviceType.GetTypeInfo().IsGenericType || serviceType.GetGenericTypeDefinition() != typeof(IEnumerable<>))
+                return;
+
+            var elementType = serviceType.GetGenericArguments().Single();
+            var array = System.Array.CreateInstance(elementType, 1);
```

Comment:
What about `IEnumerable<IEnumerable<T>>`?

Reply:
I guess resolvers would work recursively. So you'd get an array of one element containing an element of one element containing whatever T the container got.

Reply:
Not sure whether to explicitly cover this or what the expected semantics of IE<IE<T>> would be. It's just a best effort in case a class has a dependency on such a nested interface.

Reply:
This is a good call out. I think we should certainly add a test for this case. From looking at the code it appears that we would likely get slightly different behavior between the first call to `.Get` and subsequent calls (due to the caching in `_typeMap`). IMO if there is an AM instance with `(int)42` registered, I would expect `IEnumerable<IEnumberable<int>>` to return an enumerable with a single `null` item. Though I could see some rational for trying to handle the recursion.

Reply:
I added a test that asserts the current behaviour. I don't know whether that would capture the semantics you or the other maintainers have in mind for that scenario though.